documentation/chatmeleon-swagger.yaml

```diff
@@ -18,107 +18,315 @@ externalDocs:
+          description: The unique conversation ID.
+          required: true
+          schema:
+            type: string
+          example: "fdf919880bcde1e42cb3e588"
+      responses:
+        "200":
+          description: A conversation record.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Conversation"
+    post:
+      description: Create a conversation by its ID.
+      tags:
+        - conversations
+      parameters:
+        - name: conversationId
+          in: path
+          description: The unique conversation ID.
+          required: true
+          schema:
+            type: string
+          example: "fdf919880bcde1e42cb3e588"
+      responses:
+        "200":
+          description: Conversation updated successfully.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Conversation"
+      requestBody:
+        description: Conversation data to update.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Conversation"
+  /conversations/{conversationId}/messages:
+    get:
+      description: Retrieve a page of messages of a conversation identified by the conversation's ID with the support of cursor-based pagination.
+      tags:
+        - messages
+      summary: Get all messages in a conversation
+      parameters:
+        - name: conversationId
+          in: path
+          description: The unique conversation ID.
+          required: true
+          schema:
+            type: string
+          example: "fdf919880bcde1e42cb3e588"
+        - name: pageSize
+          in: query
+          description: The number of conversations in one page.
+          schema:
+            type: integer
+            default: 10
+        - name: cursor
+          in: query
+          description: The ID of the earliest message of the previous messages page.
+          schema:
+            type: string
+            pattern: "^[0-9a-f]{24}$"
+            default: null
+      responses:
+        "200":
+          description: All messages's information.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  allMessages:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/Message"
+                  nextCursor:
+                    type: string
+                    pattern: "^[0-9a-f]{24}$"
+                    description: The ID of the earliest message of the current messages page.
+  /users/{userId}/messages:
+    get:
+      description: Retrieve a page of messages of a conversation identified by the user's ID with the support of cursor-based pagination.
+      tags:
+        - messages
+      summary: Get all messages of a user
```

Comment:
What does this messages of a user do? Aren't messages belong to a conversation? @MapleThanh

Reply:
Yea, it is quite irrelevant to the current state of our app. @XkhangNguyen can you please remove this route in the doc and also in the controller